### PR TITLE
make identificaiton of closed indices more clear

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**

### CODE REVIEW

**Comment by asd-and-Rizzo** [comment]
https://elasticsearch.na1.echosign.com/public/viewAgreement?tsid=CBFCIBAA3AAABLblqZhAw9JIc_nx4zzBm6p5I0Eb3OvgGcMdYvUBhpd4WEZqdu06WW-a3-ZojdMqUmyKveZZprmKUtIftRIY6ZHbirm_a&

File: docs/reference/indices/open-close.asciidoc [hunk 1]
```diff
@@ -32,3 +32,5 @@ This setting can also be changed via the cluster update settings api.
 
 Closed indices consume a significant amount of disk-space which can cause problems in managed environments. Closing indices can be disabled via the cluster settings
 API by setting `cluster.indices.close.enable` to `false`. The default is `true`.
+
+Closed index responses with `index_closed_exception` and `403` status on call to `_stats` API. Closed indices are not returned in cross index `_aliases` or `_stats` API responses.
```
Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: there are many more APIs that return IndexClosedException and 403 status code when executed against closed indices. For instance _search. But through the `ignore_unavailable` parameter, you can decide whether to ignore those closed indices or get an exception back instead. I would rephrase your new paragraph without mentioning any specific APIs, rather say "most APIs don't support execution against closed indices, hence `ignore_unavailable allows to control whether closed indices are skipped or cause an IndexClosedException to be returned". What do you think?

**Review by javanna — CHANGES_REQUESTED** [review]
hi @asd-and-Rizzo thanks for your PR I requested a change.

**Comment by asd-and-Rizzo** [comment]
I will check `ignore_unavailable` behavior and update as recommend within next 2 weeks.

**Comment by javanna** [comment]
thanks @asd-and-Rizzo please ping me once you've done so

**Comment by asd-and-Rizzo** [comment]
@javanna, Please check my comment in patch. I see your thinking of closed behavior could be different from what I see. What does ES version your knowledge steam from?

File: docs/reference/indices/open-close.asciidoc [hunk 2]
```diff
@@ -32,3 +32,7 @@ This setting can also be changed via the cluster update settings api.
 
 Closed indices consume a significant amount of disk-space which can cause problems in managed environments. Closing indices can be disabled via the cluster settings
 API by setting `cluster.indices.close.enable` to `false`. The default is `true`.
+
+Closed index responses with `index_closed_exception` and `403` for most APIs, including `_stats` and `_search`. `ignore_unavailable` allows to control whether closed index do not cause `index_closed_exception` to be returned.  Index returns value on call to `_aliases`.
```
Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: I would slightly rephrase this sentence: "Most APIs, including `_stats` and `_search`, return `index_closed_exception` and `403` status code when accessing closed indices. The `ignore_unavailable querystring parameter allows to control whether closed indices cause an error to be returned or not.".

File: docs/reference/indices/open-close.asciidoc [hunk 3]
```diff
@@ -32,3 +32,7 @@ This setting can also be changed via the cluster update settings api.
 
 Closed indices consume a significant amount of disk-space which can cause problems in managed environments. Closing indices can be disabled via the cluster settings
 API by setting `cluster.indices.close.enable` to `false`. The default is `true`.
+
+Closed index responses with `index_closed_exception` and `403` for most APIs, including `_stats` and `_search`. `ignore_unavailable` allows to control whether closed index do not cause `index_closed_exception` to be returned.  Index returns value on call to `_aliases`. 
+
+Closed indices are not returned in most wildcard index API calls, like `_aliases`, `_stats` or `_search`, nor throw any exceptions and behave like non-existent.
```
Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: I am not sure about this sentence. You control through ignore_unavailable whether closed indices throw error or not. They are unavailable indices for most of the APIs, hence either they are ignored or they cause an error (e.g. it is not technically possible to search against a closed index, so the question is just what to do instead: error or skip them?)

File: docs/reference/indices/open-close.asciidoc (hunk 2 above)
Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: What do you mean with "Index returns value on call to `_aliases`" ?

**Review by javanna — CHANGES_REQUESTED** [review]
hi @asd-and-Rizzo I left a couple more comments. My feedback doesn't stem from any specific Elasticsearch version, ignore_unavailable is supported for a very long time now. Let me know if you have any further questions.

**Comment by asd-and-Rizzo** [comment]
Here is case:
/my_index
/my_closed_index

1. Request:
/_aliases
Response:
/my_index

2. Request:
/my_closed_index/_aliases
Response:
/my_closed_index

2. Request:
/_search
Response:
Some values from my_index
No error from my_closed_index.

3. Request:
/my_closed_index/_search?ignore_unavailable
Response:
Zero hits.

4. Request:
/my_closed_index/_search
Response:
Error.


From these I conclude next:
1. I should keep external list of indices to be able to delete them later (e.g. because of my retention policy we do close and then delete) or access ES cluster directly, not via API, to find the list.
2. ignore_unavailable does not work on multi indeces requests. So it makes `ignore_unavailable` useless in some situations.

So the problem is what is correct and useful specification of closed indices calls and usage of ignore_unavailable?

I will rephrase the sentence, but current behavior of closed indexes does not fill consistent for me.

**Comment by javanna** [comment]
hi @asd-and-Rizzo  we'd like users to ask questions in our [forums](https://discuss.elastic.co/), as we reserve github issues for verified bug reports and feature requests.

From your example I see that `ignore_unavailable` does work with e.g. the search API. But I am not clear on the following statement:

> ignore_unavailable does not work on multi indeces requests. So it makes ignore_unavailable useless in some situations.

Please do open another issue if you think that the above is a bug rather than commenting here.

This other issue may be interesting too: #9438

**Comment by asd-and-Rizzo** [comment]
@javanna, thanks for reference to forum and other issues. My pull request was only about to add documentation about current behavior, no expectancy to change this behavior. Just helping our support team and hoping that current behavior inconsistencies(as I see these) will be visible to other people.

File: docs/reference/indices/open-close.asciidoc [hunk 4]
```diff
@@ -32,3 +32,10 @@ This setting can also be changed via the cluster update settings api.
 
 Closed indices consume a significant amount of disk-space which can cause problems in managed environments. Closing indices can be disabled via the cluster settings
 API by setting `cluster.indices.close.enable` to `false`. The default is `true`.
+
+Most APIs, including _stats and _search, return `index_closed_exception` and 403 status code when accessing closed index.  The `ignore_unavailable` querystring parameter allows to control whether closed index cause an error to be returned or not.  Call to index `_aliases` returns correct value without error.
```
Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: 403 recently became 400 . Can you update that?

Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: when accessing closed ind**ices**

Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: whether closed ind**ices** cause an error to be returned or are skipped.

Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: I find this sentence confusing: "Call to index `_aliases` returns correct value without error. " I would remove it.

File: docs/reference/indices/open-close.asciidoc [hunk 5]
```diff
@@ -32,3 +32,10 @@ This setting can also be changed via the cluster update settings api.
 
 Closed indices consume a significant amount of disk-space which can cause problems in managed environments. Closing indices can be disabled via the cluster settings
 API by setting `cluster.indices.close.enable` to `false`. The default is `true`.
+
+Most APIs, including _stats and _search, return `index_closed_exception` and 403 status code when accessing closed index.  The `ignore_unavailable` querystring parameter allows to control whether closed index cause an error to be returned or not.  Call to index `_aliases` returns correct value without error. 
+
+Closed indices are not returned in most wildcard API calls, like `_aliases`, `_stats` or `_search`, nor throw any exceptions. Closed indices behave like non-existent, closed indices will not be returned during call to `_aliases`. Passing `ignore_unavailable` is irrelevant to wildcard indices calls.
```
Comment (javanna) [inline docs/reference/indices/open-close.asciidoc]: expand_wildcards allows to control whether to expand wildcards against open, closed or both. can you update this sentence please?

**Review by javanna — CHANGES_REQUESTED** [review]
sorry @asd-and-Rizzo it took me a while to have another look, I left some more comments. Could you address those so I can get your PR in?

**Comment by javanna** [comment]
There has been no activity on this for two months, I am closing. Feel free to reopen once the comments are addressed.
